java/dataset/src/main/java/org/apache/arrow/dataset/scanner/Scanner.java

```diff
@@ -17,19 +17,29 @@
 
 package org.apache.arrow.dataset.scanner;
 
+import org.apache.arrow.vector.ipc.ArrowReader;
 import org.apache.arrow.vector.types.pojo.Schema;
 
 /**
  * A high level interface for scanning data over dataset.
  */
 public interface Scanner extends AutoCloseable {
 
+  /**
+   * Read record batches from the URI resources filtering the batch size configured.
+   *
+   * @return a {@link ArrowReader}.
+   */
+  ArrowReader scanBatchesUnordered();
```

Comment:
Sorry, it looks like it's actually `scanBatches` in the implementation: https://github.com/apache/arrow/blob/2629f208cf8be638c7e251946947009bc181bb31/java/dataset/src/main/cpp/jni_wrapper.cc#L152

Reply:
Thanks, updated